src/Compiler/PersistentSystemWebCompilation.cs

```diff
@@ -0,0 +1,105 @@
+// MIT License.
+
+using System.Globalization;
+using System.Text.Json;
+using System.Web.UI;
+using Microsoft.CodeAnalysis;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace WebForms.Compiler;
+
+internal sealed class PersistentSystemWebCompilation : SystemWebCompilation, IWebFormsCompiler
+{
+    private readonly IOptions<PersistentCompilationOptions> _options;
+    private readonly IOptions<PageCompilationOptions> _pageOptions;
+    private readonly ILogger _logger;
+
+    public PersistentSystemWebCompilation(
+        IOptions<PersistentCompilationOptions> options,
+        IOptions<PageCompilationOptions> pageOptions,
+        IOptions<PagesSection> pagesSection,
+        IOptions<CompilationSection> compilationSection,
+        ILoggerFactory factory)
+        : base(factory, pagesSection, compilationSection)
+    {
+        _logger = factory.CreateLogger<PersistentSystemWebCompilation>();
+        _options = options;
+        _pageOptions = pageOptions;
+    }
+
+    protected override ICompiledPage CreateCompiledPage(Compilation compilation, string route, string typeName, IEnumerable<SyntaxTree> trees, IEnumerable<MetadataReference> references, IEnumerable<EmbeddedText> embedded, CancellationToken token)
```

Comment:
we don't need metadata references (just like the dynamic compilation) ? If not, could you please provide bit info for my understanding...

Reply:
The dynamic compilation is using the existing referenced assemblies; we need something to compile against. There's definitely cleanup warranted here, it's mostly to get something in place that can emit assemblies